lib/sincerely/services/process_delivery_event.rb

```diff
@@ -50,20 +52,18 @@ def call
       attr_reader :event
 
       def process_bounce_event
-        notification.set_bounced!
-        Sincerely::BounceEvent
-          .create(message_id:, delivery_system:, recipient:, timestamp:, options:)
+        Sincerely::DeliveryEvent.create(message_id:, delivery_system:, event_type: :bounce, recipient:, timestamp:,
+                                        options:)
       end
 
       def process_complaint_event
-        notification.set_complained!
-        Sincerely::ComplaintEvent
-          .create(message_id:, delivery_system:, recipient:, timestamp:, options:)
+        Sincerely::DeliveryEvent.create(message_id:, delivery_system:, event_type: :complaint, recipient:, timestamp:,
+                                        options:)
       end
 
       def process_engagement_event
-        Sincerely::EngagementEvent
-          .create(message_id:, delivery_system:, recipient:, timestamp:, options:)
+        Sincerely::DeliveryEvent.create(message_id:, delivery_system:, event_type: :engagement, recipient:, timestamp:,
```

Comment:
I think event_type should be the actual event type here (open/click) just to be able to distinguish them